src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ModifiedControlVariableCheck.java

```diff
@@ -289,13 +287,11 @@ private void exitBlock() {
      */
     private void checkIdent(DetailAST ast) {
         final Deque<String> currentVariables = getCurrentVariables();
-        if (currentVariables != null && !currentVariables.isEmpty()) {
```

Comment:
It was only `null` in weird places like in the middle of an annotation. `@Example(value = "")`.
An empty list is just an extra pre-caution from doing more work before calling `contains`.